var/spack/repos/builtin/packages/sollve/package.py

```diff
@@ -0,0 +1,242 @@
+    depends_on('perl-data-dumper', type=('build'))
+    depends_on('argobots', when='+argobots')
+
+    # lldb dependencies
+    depends_on('ncurses', when='+lldb')
+    depends_on('swig', when='+lldb')
+    depends_on('libedit', when='+lldb')
+    depends_on('py-six', when='+lldb +python')
+
+    # gold support
+    depends_on('binutils+gold', when='+gold')
+
+    version("develop", git='https://github.com/SOLLVE/llvm.git')
+    resource(name='compiler-rt',
+             svn='http://llvm.org/svn/llvm-project/compiler-rt/trunk',
+             destination='projects', when='@develop+compiler-rt',
+             placement='compiler-rt')
+    resource(name='openmp', git='https://github.com/pmodels/bolt.git',
+             destination='projects', when='@develop+clang', placement='openmp')
+    resource(name='polly', git='https://github.com/SOLLVE/polly.git',
+             destination='tools', when='@develop+polly', placement='polly')
+    resource(name='libcxx', git='https://github.com/SOLLVE/libcxx.git',
+             destination='projects', when='@develop+libcxx',
+             placement='libcxx')
+    resource(name='libcxxabi', git='https://github.com/SOLLVE/libcxxabi.git',
+             destination='projects', when='@develop+libcxx',
+             placement='libcxxabi')
+    resource(name='cfe', git='https://github.com/SOLLVE/clang.git',
+             destination='tools', when='@develop+clang', placement='clang')
+    resource(name='lldb', svn='http://llvm.org/svn/llvm-project/lldb/trunk',
+             destination='tools', when='@develop+lldb', placement='lldb')
+    resource(name='lld', svn='http://llvm.org/svn/llvm-project/lld/trunk',
+             destination='tools', when='@develop+lld', placement='lld')
+    resource(name='libunwind',
+             svn='http://llvm.org/svn/llvm-project/libunwind/trunk',
+             destination='projects', when='@develop+internal_unwind',
+             placement='libunwind')
+
+    conflicts('+clang_extra', when='~clang')
+    conflicts('+lldb',        when='~clang')
+
+    conflicts('%gcc@:5.0.999')
+    conflicts('+omp_tsan')
+
+    @run_before('cmake')
+    def check_darwin_lldb_codesign_requirement(self):
+        if not self.spec.satisfies('+lldb platform=darwin'):
+            return
+        codesign = which('codesign')
+        mkdir('tmp')
+        llvm_check_file = join_path('tmp', 'llvm_check')
+        copy('/usr/bin/false', llvm_check_file)
+
+        try:
+            codesign('-f', '-s', 'lldb_codesign', '--dryrun',
+                     llvm_check_file)
+
+        except ProcessError:
+            explanation = ('The "lldb_codesign" identity must be available'
+                           ' to build LLVM with LLDB. See https://llvm.org/'
+                           'svn/llvm-project/lldb/trunk/docs/code-signing'
+                           '.txt for details on how to create this identity.')
+            raise RuntimeError(explanation)
+
+    def setup_environment(self, spack_env, run_env):
```

Comment:
Needs to be split into `setup_build_environment` and `setup_run_environment` to get the unit tests to pass

Reply:
I split them and create two functions `setup_build_environment` and `setup_run_environment`.